### PR TITLE
add mystran

### PR DESCRIPTION
**Pull request by looooo**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by looooo** [comment]
https://github.com/MYSTRANsolver/MYSTRAN/issues/83
https://github.com/MYSTRANsolver/MYSTRAN/issues/82

**Comment by looooo** [comment]
@conda-forge/core ready for review. Superlu is build within this recipe see this issue: https://github.com/MYSTRANsolver/MYSTRAN/issues/82 .

File: recipes/mystran/build.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+cmake -B build -S . \
+      -D CMAKE_BUILD_TYPE="Release" \
+      -D CMAKE_INSTALL_PREFIX:FILEPATH=$PREFIX
+
+make -C build install
+mkdir -p $PREFIX/bin
```
Comment (ocefpaf) [inline recipes/mystran/build.sh:6]: Any make check or ctest that we can run here?

Comment (looooo) [reply recipes/mystran/build.sh:6]: not sure if there is much testing available. I added some test files which run with mystran.

**Comment by looooo** [comment]
The test files are still not available:
(%PREFIX%) %SRC_DIR%>mystran Build_Test_Cases/statics/statics/bar_tube.bdf 
 Optional Initialization file: 
```
 MYSTRAN.INI
 does not exist. Internal defaults will be used instead.                        
 If you want to use the INI file, see the MYSTRAN Installation and Run Manual   
```
Not sure if I have done the copying fron the source dir correctly: 
https://github.com/looooo/staged-recipes/blob/mystran/recipes/mystran/meta.yaml#L36L38

File: recipes/mystran/build.sh (hunk 1 above)
Comment (ocefpaf) [reply recipes/mystran/build.sh:6]: If you believe what you have here good enough to test it, we can merge it.

Comment (looooo) [reply recipes/mystran/build.sh:6]: Yes, maybe there will be more tests avaoilable in the future

**Pull request merged**